src/FastPriorityQueue/PHP81Implementation.php

```diff
@@ -0,0 +1,431 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Stdlib\FastPriorityQueue;
+
+use Countable;
+use Iterator;
+use Laminas\Stdlib\Exception;
+use Serializable;
+use SplPriorityQueue as PhpSplPriorityQueue;
+use UnexpectedValueException;
+
+use function array_key_exists;
+use function current;
+use function get_class;
+use function gettype;
+use function in_array;
+use function is_array;
+use function is_int;
+use function is_object;
+use function key;
+use function max;
+use function next;
+use function reset;
+use function serialize;
+use function sprintf;
+use function unserialize;
+
+/**
+ * This is an efficient implementation of an integer priority queue in PHP
+ *
+ * This class acts like a queue with insert() and extract(), removing the
+ * elements from the queue and it also acts like an Iterator without removing
+ * the elements. This behaviour can be used in mixed scenarios with high
+ * performance boost.
+ */
+class PHP81Implementation implements Iterator, Countable, Serializable
```

Comment:
Should be `@internal` IMO: the implementation should not be part of the public API